Dockerfile

```diff
@@ -1,27 +1,44 @@
-# Dockerfile
-# ---- Base Node ----
+# Base stage
 FROM node:20.11.0 AS base
-ENV NEXT_TELEMETRY_DISABLED 1
-WORKDIR /app
-COPY apps/web-staking/package.json ./apps/web-staking/
-COPY apps/web-staking/package-lock.json ./apps/web-staking/
-RUN npm install --loglevel verbose -dd --prefix ./apps/web-staking
 
-# ---- Build ----
-FROM base AS build
-ENV NEXT_TELEMETRY_DISABLED 1
+# Set environment variables
+ENV NEXT_PUBLIC_APP_ENV=development
+
+# Install pnpm globally
+RUN npm install -g pnpm
+
+# Set the working directory inside the container
 WORKDIR /app
+
+# Copy the entire monorepo to the container, excluding node_modules and other ignored files
 COPY . .
-RUN npm run build --prefix ./apps/web-staking
 
-# --- Release ----
-FROM gcr.io/distroless/nodejs20-debian11 AS release
-ENV NEXT_TELEMETRY_DISABLED 1
+# Install all dependencies for the monorepo
+RUN pnpm install
+
+# Build the web-staking application
+WORKDIR /app/apps/web-staking
+RUN pnpm build
+
+# Release stage
+FROM node:20.11.0 AS release
+
+# Set environment variables
+ENV NEXT_PUBLIC_APP_ENV=development
+
+# Install pnpm globally
+RUN npm install -g pnpm
+
+# Set the working directory inside the container
 WORKDIR /app
-COPY --from=build /app/apps/web-staking/.next ./.next
-COPY --from=build /app/apps/web-staking/public ./public
-COPY --from=base /app/apps/web-staking/node_modules ./node_modules
-COPY apps/web-staking/package.json .
-COPY apps/web-staking/package-lock.json .
+
+# Copy only necessary files from the base stage
+COPY --from=base /app/apps/web-staking /app/apps/web-staking
+COPY --from=base /app/node_modules /app/node_modules
+COPY --from=base /app/package.json /app/package.json
+
+# Expose the port the app runs on
 EXPOSE 3000
-CMD ["./node_modules/next/dist/bin/next", "start"]
+
+# Command to run the application
+CMD ["pnpm", "staking"]
```

Comment:
Im not sure about starting with pnpm right here. This would require an update on the deployment pipeline from the services team. We should try to find a way not breaking the current deployment